### DISCUSSION TITLE
Update choices of a choice field

### DISCUSSION THREAD

**Discussion by OctavianSfetcu** [pnp/pnpcore#1003] [Q&A] [answered]
Hi

how can I update the choices of a choice field? I tried to update the choices of a choice field in a document library with the following code:

var choiceField = docLib.Fields.AsRequested().FirstOrDefault(f => f.InternalName == syncItem.ChoiceField);
choiceField.Choices = new string[] { "Choice001", "Choice002", "Choice003" };
await choiceField.UpdateAsync();

But I get the error:

**The Choices property does not exist for the SP.Field type. Make sure to only use property names that are defined in the type.**

When I check this choiceField.Choices, the original choices are there.

Thanks.

**Comment by jansenbe** [comment, marked as answer]
@OctavianSfetcu : can you try by updating the `SchemaXml` and then calling `UpdateAsync`?

**Reply by OctavianSfetcu** [reply]
Hey @jansenbe 
it worked! Thank you so much for the advice!
Is this a bug? Should also work if I set the Choices in the field, right?
All the best
Octavian

**Reply by jansenbe** [reply]
@OctavianSfetcu : one can argue if it's a bug or not, to support such a scenario we would need to deserialize the received SchemaXml to the respective properties and on update do the reverse. The issue however is that the schema XML is quite extensive, and we simply don't support each and every possible option/combination as property, hence it's much safer for the library users to update an existing XML definition and send that back as update.
